### ISSUE TITLE
Levelling/personal cash system

### ISSUE THREAD

**Issue by Inf3rn0666** [Wyqer/kp_liberation#327]
Me and a few mates are playing the crap out of this mod. we were thinking it would be cool to have a sort of leveling system where higher tier weapons are restricted to certain levels similar to how king of the hill does it.

**Comment by Wyqer** [comment]
I would like to see more opinions on this.
Personally I would say "Liberation is Liberation" where you're a regular army force, which shouldn't have to "unlock" their personal weapons. Unlocking vehicles is something "grey" for me.... could be explained by "as the operation is progressing well, we'll grant you access to much more expensive vehicles for your tasks"...... but also for Sniper Rifle X or Rocket Launcher Y. I guess this should stay at a KOTH styled mission.
But I'm not the only one, who can state his opinion here.

**Comment by runy888** [comment]
Supply chain and the availability of advanced equipment is an issue in real conflicts as well. Especially with the level of customisation that your mission offers many scenarios can be represented. Admittedly, most versions of Liberation see the "West" fighting the "East" but with just little change it can turn into an insurgency or civil war style mission. Would that defeat the purpose you see in it? 
Adding an arsenal that increases in strength and quality as the campaign progresses would be an excellent addition in my opinion, and it would mark the stepping stone to making supply chain issues actually impact the equipment players have available. I'm thinking having to install infrastructure on FOBs to add pieces of equipment to its arsenal etc. But that's going quite far, for the start a scaling arsenal would extend the mission massively in a whole different direction.

**Comment by SondreBakken** [comment]
Something I really wish to see, that this could help with is benefits of staying alive. 

Right now I see people suicide because no-one wanna do a heli pick-up and its to far to walk.

In a game where you fight to stay alive, but there is no comparing of stats etc or other incentives to stay alive kinds of make you detached from your avatar. He is just an expandable soldier. Giving people a reason to stay alive would greatly increase immersion. 

So something that is directly connected to the soldier you plays, that is reset on death and increased through effort and good skills.

So some experience stat, or reputation stat. But what benefits would this give you that doesn't change liberation gameplay to much.

Maybe you could have a hidden exprience stat that would give you rank. 
Only your rank is visible, so for example: 0-200 - private, 200-600: leutenant, 600+ captain and so on.
The different ranks give you a discount when buying stuff. So at private, no discount, at captain
you could get a 30% discount. 

This would create a incentive to stay alive, but also keeping others alive.
Ofc, you might experience that the guy with the highest rank would buy stuff for others, but I guess thats ok. It would still be beneficial to have a high rank yourself. Also the ones with really high rank would be more worth protecting because a General that might have 50% discount would impact the whole game so if he goes down its reeaaaally bad :) Maybe opfor could target people with really high ranks.

**Comment by Blakmane** [comment]
As a general concept I can't say I'm a fan of individual equipment unlocks. From my understanding this map isn't meant to be persistent between playthroughs, and it also makes no sense in the context of an 'army'-style campaign. It would also be frustrating for drop-in players or friends. It's also potentially super annoying given the current loadouts menu as you'd have to constantly adjust/make new loadouts as you get access to more gear.  I could see some type of 'perk' unlocks similar to DUWS though (the ability to remotely call in a paradrop, chopper EVAC etc using the radio menu).

The one thing I will say is that some high tier equipment e.g. titan launchers and IR goggles dominate play especially early on, so it could be nice to limit access or require resources to purchase. However, I also feel like that might end up just being more of a hassle than it is worth.

**Comment by dmorchard** [comment]
Here's all the incentive I would need to stay alive: a stamina modifier that reduces the associated effects (sprint duration, weapon sway, etc) over time but resets whenever the player dies or forces a respawn.   I confess I have no idea if this is simple, complex or impossible to implement :)

**Comment by AngusDLX** [comment]
For what it seems, giving "rewards" greatly increases team performance especially when doing boring tasks like logistics (I know it can be automated). I did an experiment with my clan friends and they really got a lot more excited when I created some tasks straight from zeus using the custom objective. We are developing a website where we have a rank showing the number of tasks completed, medals for performance in battle, etc.

A system where soldiers would unlock weapons according to the rewards accumulated in my view could be a very good thing but it can be something optional in Liberation.
